### ISSUE TITLE
Doesn't work with Scala 2.13 or 3

### ISSUE THREAD

**Issue by eed3si9n** [hamnis/dataclass-scalafix#15]
Ref https://github.com/hamnis/dataclass-scalafix/issues/4

## steps

Use Scala 3

```scala
package lmcoursier.definitions

import dataclass._

@data class Strict(
  include: Set[(String, String)] = Set(("*", "*")),
  exclude: Set[(String, String)] = Set.empty,
) {
}
```

## problems

```scala
[error] -- [E164] Declaration Error: /Users/xxx/work/sbt-coursier/modules/lm-coursier/target/scala-3.1.3/src_managed/main/scala/lmcoursier/definitions/Strict.scala:37:4
[error] 37 |def productElementName(n: Int) = n match {
[error]    |    ^
[error]    |error overriding method productElementName in trait Product of type (n: Int): String;
[error]    |  method productElementName of type (n: Int): String needs `override` modifier
[error] -- [E164] Declaration Error: /Users/xxx/work/sbt-coursier/modules/lm-coursier/target/scala-3.1.3/src_managed/main/scala/lmcoursier/definitions/Strict.scala:51:4
[error] 51 |def productElementNames = {
[error]    |    ^
[error]    |error overriding method productElementNames in trait Product of type => Iterator[String];
[error]    |  method productElementNames of type => Iterator[String] needs `override` modifier
```

## notes

Using Scalafix from sbt, I am guessing that the rule will always use the same version as the metabuild, in this case Scala 2.12.
I don't know if there's a way to detect that the target dialect should be Scala 3.
https://github.com/scalacenter/scalafix/blob/main/scalafix-core/src/main/scala/scalafix/v1/SemanticDocument.scala doesn't seem to contain Scala version.

**Comment by hamnis** [comment]
Is it possible to to use scalafixScalaBinaryVersion

**Comment by eed3si9n** [comment]
Nice. I can do that.

```scala
ThisBuild / scalafixScalaBinaryVersion := {
  (ThisBuild / scalaBinaryVersion).value match {
    case "3" => "2.13"
    case v   => v
  }
}
```

`3` errors out saying some rules don't support it, but thankfully 2.13 code was source compatible.

**Comment by bjaglin** [comment]
You don't need to use `scalafixScalaBinaryVersion` as your rule (whether it's running with 2.12 or 2.13) can retrieve the scala version of the source files it runs against (through the `--scala-version` argument in [scalafix](https://scalacenter.github.io/scalafix/docs/users/installation.html#help), automatically injected by sbt-scalafix based on the `scalaVersion` setting key) and adapt accordingly.

See https://github.com/hamnis/dataclass-scalafix/issues/4#issuecomment-1212034756.

**Comment by eed3si9n** [comment]
Thank you! That sounds great.

**Comment by hamnis** [comment]
Fixed by #16
